src/main/java/onlineservices/services/WindowControl/WindowControlService.java

```diff
@@ -1,16 +1,50 @@
 package onlineservices.services.WindowControl;
 
+import org.eclipse.paho.client.mqttv3.*;
 import onlineservices.OnlineService;
 
 public class WindowControlService implements OnlineService {
 
+    private static final String MQTT_BROKER_URL = "tcp://broker.emqx.io:1883";
+    private static final String MQTT_CLIENT_ID = "WindowControlService";
+    private static final String MQTT_WINDOW_SERVICE_TOPIC = "WindowControlService";
+    private MqttClient client;
+
     @Override
     public void onCreate() {
```

Comment:
create a new class called MqttClientHandler

this class will have the following methods:
connect(required connection variables)
disconnect()
subscribeToTopic(topic, responseHandler)
unsubscribeFromTopic(topic, any other required variale)
basically i want a wrapper for the mqtt client and i want any operations that you want to perform on the mqtt client to be encapsulated in this class.

put this:
client = new MqttClient(MQTT_BROKER_URL, MQTT_CLIENT_ID);
in the constructor of the MqttClientHandler

put this:
client.connect();
in the connect() method

put this
client.disconnect();
in the disconnect method

create a new class for the Callback, call it WindowControlRequestCallback, class should extend MqttCallback, or implement it, not sure if it's an interface or class

structure the classes in appropriate packages, example: handlers, controllers, services, models, utils, etc...